File: c/c99/1040_media_3.c
#include <stdio.h>

int main() {
    float N1, N2, N3, N4, media, exame, media_final;
    scanf("%f %f %f %f", &N1, &N2, &N3, &N4);
    media = ((N1*2)+(N2*3)+(N3*4)+(N4*1))/(2+3+4+1);
    printf("Media: %.1f\n", media);
    if (media >= 7.0) {
        printf("Aluno aprovado.\n");
    }
    else if (media < 5.0) {
        printf("Aluno reprovado.\n");
    } else {
        printf("Aluno em exame.\n");
        scanf("%f", &exame);
        printf("Nota do exame: %.1f\n", exame);
        media_final = (exame+media)/2;
        if (media_final >= 5.0) {
            printf("Aluno aprovado.\n");
        } else {
            printf("Aluno reprovado.\n");
        }
        printf("Media final: %.1f\n", media_final);
    }
}

File: python/1040_media_3.py
import math

N1, N2, N3, N4 = map(float, input().split())
P1 = 2
P2 = 3
P3 = 4
P4 = 1

MEDIA = ((N1*P1)+(N2*P2)+(N3*P3)+(N4*P4))/10
MEDIA_REAL = math.floor(MEDIA*10)/10
print(f'Media: {MEDIA_REAL:.1f}')

if MEDIA >= 7.0:
    print(f'Aluno aprovado.')
elif MEDIA < 5.0:
    print(f'Aluno reprovado.')
else:
    print(f'Aluno em exame.')
    A = float(input())
    print(f'Nota do exame: {A}')
    MEDIA_F = (MEDIA+A)/2
    if MEDIA_F >= 5.0:
        print(f'Aluno aprovado.')
        print(f'Media final: {MEDIA_F:.1f}')
    else:
        print(f'Aluno reprovado.')
        print(f'Media final: {MEDIA_F:.1f}')
